### PR TITLE
msg/async/ProtocolV2: do not reenable pre_auth buffering on from reset_recv_state

### PR DESCRIPTION
**Pull request by liewegas**
This is specifically bad because we call reset_recv_state from
reuse_connection, which turns buffering back on on an already-authenticated
session.

Instead, reenable it only when we set the state to START_CONNECT.  (On
the accepting side, it is a fresh connection, so it starts out true.)

Fixes: http://tracker.ceph.com/issues/38746
Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

**Comment by rzarzynski** [comment]
http://pulpito.front.sepia.ceph.com/rzarzynski-2019-03-15_06:25:40-rados-wip-sage-testing-2019-03-14-2300-distro-basic-smithi/

**Comment by rjfd** [comment]
also reenable pre_auth around line 370 (https://github.com/ceph/ceph/pull/26965/files#diff-ea445419e9f7f9529b73ce454e5e3c53R370)?

**Comment by liewegas** [comment]
updated

File: src/msg/async/ProtocolV2.cc [hunk 1]
```diff
@@ -2626,6 +2628,7 @@ CtPtr ProtocolV2::reuse_connection(AsyncConnectionRef existing,
     ceph_assert(!connection->delay_state);
   }
   exproto->reset_recv_state();
+  exproto->pre_auth.enabled = false;
```
Comment (rjfd) [inline src/msg/async/ProtocolV2.cc]: also add `exproto->pre_auth.rxbuf.clear();` below this line

Comment (liewegas) [reply src/msg/async/ProtocolV2.cc]: that is already done by the reset_recv_state, right?

Comment (rjfd) [reply src/msg/async/ProtocolV2.cc]: you're right, disregard this comment

**Pull request merged**